osu.Game/Screens/Menu/IntroTriangles.cs

```diff
@@ -78,6 +88,12 @@ public override void OnResuming(IScreen last)
             background.FadeOut(100);
         }
 
+        protected override void StartTrack()
+        {
+            if (UsingThemedIntro)
+                decoupledClock.Start();
+        }
```

Comment:
As `TrianglesIntroSequence` now relies on the decoupled clock regardless of `UsingThemedIntro`, not starting the clock here would leave intro frozen in the `UsingThemedIntro = false` case (can confirm by hardcoding that property to that value).

It should be safe to start the decoupled clock in all cases as it doesn't source from `Track` until `UsingThemedIntro = true` anyways.
```suggestion
        protected override void StartTrack() => decoupledClock.Start();
```

Reply:
is there an actual feasible, reproducible in-game scenario in which that can happen, that doesn't involve hardcoding random values? because i can't find one, and i've tried removing all beatmaps and disabling intro theme.

as far as i can see `UsingThemedIntro` is pretty much guaranteed to be `true` for all intro types. to the point that i'm not even entirely sure what purpose it serves.

Reply:
Looking at the code behind it, it must be the case when you have `OsuSetting.MenuMusic` turned off and the randomly selected beatmap is one that has a virtual track backing it. Didn't bother to actually have all of that set up and went with the easiest way of hardcoding that.

Reply:
that is a better way of describing the potential failure scenario. and does indeed reproduce.

Reply:
Interesting. May be a good chance to make this more obvious, because the handling in the existing code made me also believe this was an impossible scenario.

Reply:
I've done a bit of refactoring to hopefully make this flow easier to follow. Not sure if this warrants adding test coverage (might be a bit tricky to implement proper testing).